tests/test_streaming.py

```diff
@@ -59,6 +59,203 @@ def test_dataloader_epoch_size_no_streams(local_remote_dir: Tuple[str,
+
+    stream1 = Stream(local=local1, remote=remote1)
+    stream2 = Stream(local=local2, remote=remote2)
+
+    # Build StreamingDataset
+    dataset = StreamingDataset(streams=[stream1, stream2],
+                               shuffle=shuffle,
+                               batch_size=batch_size,
+                               shuffle_seed=seed,
+                               num_canonical_nodes=num_canonical_nodes,
+                               batching_method='stratified')
+
+    # Build DataLoader
+    dataloader = DataLoader(dataset=dataset,
+                            batch_size=batch_size,
+                            num_workers=num_workers,
+                            drop_last=drop_last)
+
+    # Ensure that the samples seen in each batch are proportional to the stream sizes.
+    total_samples = num_stream_1_samples + num_stream_2_samples
+    stream_1_batch_part = round(batch_size * (num_stream_1_samples / total_samples))
+    stream_2_batch_part = batch_size - stream_1_batch_part
+
+    # The total number of possible batches is the minimum of the batch parts from each stream.
+    # Total number of samples will be padded to be divisible by NCN. This logic takes care of that case.
+    total_stream_1_batches = num_stream_1_samples // stream_1_batch_part if num_stream_1_samples % num_canonical_nodes == 0 else (
+        num_stream_1_samples +
+        (num_canonical_nodes - num_stream_1_samples % num_canonical_nodes)) // stream_1_batch_part
+    total_stream_2_batches = num_stream_2_samples // stream_2_batch_part if num_stream_2_samples % num_canonical_nodes == 0 else (
+        num_stream_2_samples +
+        (num_canonical_nodes - num_stream_2_samples % num_canonical_nodes)) // stream_2_batch_part
+    total_batches = min(total_stream_1_batches, total_stream_2_batches)
+    batches_seen = 0
+    for batch in dataloader:
+        batches_seen += 1
+        samples = batch['sample']
+        # Check if constructed batch is the correct size
+        assert len(samples) == batch_size
+        stream_1_samples = 0
+        stream_2_samples = 0
+        for sample in samples:
+            # stream 1 goes until num_stream_1_samples*3, stream 2 is everything after
+            if sample < num_stream_1_samples * 3:
+                stream_1_samples += 1
+            else:
+                stream_2_samples += 1
+        # check that the batch is consistently composed of the correct number of samples from each stream
+        assert stream_1_samples == stream_1_batch_part
+        assert stream_2_samples == stream_2_batch_part
+
+    # Check if the number of batches seen is correct
+    assert batches_seen == total_batches
+
+
+@pytest.mark.parametrize('batch_size', [4, 7])
+@pytest.mark.parametrize('seed', [2222])
+@pytest.mark.parametrize('shuffle', [True])
+@pytest.mark.parametrize('drop_last', [True])
+@pytest.mark.parametrize('num_workers', [1])
+@pytest.mark.parametrize('num_canonical_nodes', [4, 8])
```

Comment:
Let just keep `8` or any number you like to reduce the number of test permutations. Each test we add, it increases the overall unit test time. Hence, we want to keep it under the limit but at a same time, ensure, our test coverage is enough.